hs-bindgen/src-internal/HsBindgen/Frontend/Macro/Reparse/Type.hs

```diff
@@ -29,9 +29,10 @@ primTypeKeyword = choice [
     -- TODO: check that "#define bool" is actually defined to _Bool
     -- it would be better to not special-case this
     , keyword "bool", identifier "bool" -- newer LLVM treat "bool" as keyword, older as identifier
+    , keyword "_Complex"
+    , keyword "complex", identifier "complex"
 
     -- The following are unsupported
-    , keyword "_Complex"
     , keyword "_Decimal32"
```

Comment:
Add a comment explaining we don't consider Complex to be a primitive type in bindgen. 

Also say that the macro reparsing infrastructure should ideally support complex numbers, in order to support things like this:

```c
#define Length int

Length f(double complex n);
```